### PR TITLE
root-signing: update maintainer and required checks

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

Updates root-signing to 
(1) add maintainer: @kommendorkapten from GitHub has been helping extensively with reviewing the code and understands the flow well enough to make changes. Maintainer diversity!
(2) update required status checks: please check me on whether I named them correct.. I named acoording to how they show up on the CI, e.g. https://github.com/sigstore/root-signing/pull/448

### CODE REVIEW

**Comment by haydentherapper** [comment]
As per community guidelines for approving new maintainers, LGTM for adding Fredrik!

File: github-sync/github-data/repositories.yaml [hunk 1]
```diff
@@ -1137,6 +1139,10 @@ repositories:
         requiredApprovingReviewCount: 1
         statusChecks:
           - DCO
+          - CI Test / yamllint
+          - CI Test / test
+          - CI Test / lint
+          - Cosign tests / validate
```
Comment (joshuagl) [inline github-sync/github-data/repositories.yaml:1145]: Do we not want "TUF Client tests / client" here?

File: github-sync/github-data/repositories.yaml [hunk 2]
```diff
@@ -1120,6 +1120,8 @@ repositories:
     collaborators:
       - username: haydentherapper
         permission: maintain
+      - username: kommendorkapten
+        permission: maintain
```
Comment (joshuagl) [inline github-sync/github-data/repositories.yaml:1124]: 💯

File: github-sync/github-data/repositories.yaml (hunk 1 above)
Comment (asraa) [reply github-sync/github-data/repositories.yaml:1145]: YES! That was the one I was peeved by in particular :P

**Comment by asraa** [comment]
ping please!

**Review by bobcallaway — APPROVED** [review]
lgtm!

**Review by cpanato — APPROVED** [review]

**Pull request merged**